cli/tools/upgrade.rs

```diff
@@ -79,8 +79,7 @@ pub async fn upgrade(upgrade_flags: UpgradeFlags) -> Result<(), AnyError> {
       };
 
       let current_is_most_recent = if upgrade_flags.canary {
-        let mut latest_hash = latest_version.clone();
-        latest_hash.truncate(7);
+        let latest_hash = latest_version.clone();
         crate::version::GIT_COMMIT_HASH == latest_hash
```

Comment:
Looks correct to me. `GIT_COMMIT_HASH` is `git rev-list -1 HEAD` and latest_hash is from https://dl.deno.land/canary-latest.txt which are both full hashes.